### PR TITLE
feat: add close() API

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Fixes: https://github.com/googleapis/gapic-generator-typescript/issues/68

### CODE REVIEW

File: typescript/test/testdata/showcase/src/v1beta1/echo_client.ts.baseline [hunk 1]
```diff
@@ -578,4 +586,16 @@ export class EchoClient {
       callSettings
     );
   }
+
+  /**
+   * Terminate the GRPC channel and close the client.
+   *
+   * The client will no longer be usable and all future behavior is undefined.
+   */
+  close(): void {
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply typescript/test/testdata/showcase/src/v1beta1/echo_client.ts.baseline]: The method we are invoking is synchronous: https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/client.ts#L90

[the remark this answers is not in the document]
Comment (merlinnot) [reply typescript/test/testdata/showcase/src/v1beta1/echo_client.ts.baseline]: Yes, but `this._{{ service.name.toCamelCase() }}Stub` is a Promise and we use it here.

[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply typescript/test/testdata/showcase/src/v1beta1/echo_client.ts.baseline]: Hm, I was hoping close() could be synchronous, but you are right. I updated the PR.

[the remark this answers is not in the document]
Comment (alexander-fenster) [reply typescript/test/testdata/showcase/src/v1beta1/echo_client.ts.baseline]: Unfortunately, due to the asynchronous nature of the constructor (it fires some promises to initialize the stubs), all member functions working with a stub must be asynchronous (must wait on the stubs to be complete).

**Comment by alexander-fenster** [comment]
@schmidt-sebastian Can you run `npm run baseline` to update the generated baselines and add that changes to this PR?

**Comment by alexander-fenster** [comment]
@schmidt-sebastian Seems like the generated code cannot compile:

image

If you run `npm test`, the `.test-showcase-out` will contain a generated client library for showcase API that you should be able to compile.

**Comment by schmidt-sebastian** [comment]
I just learned a lot about this tooling :) The generated code should compile now, but it doesn't since we need to export the APICallback type. I'll create a PR for that in google-gax.

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 2]
```diff
@@ -589,4 +602,19 @@ export class {{ service.name }}Client {
...
 {%- endfor %}
 {%- endif %}
+
+  /**
+   * Terminate the GRPC channel and close the client.
+   *
+   * The client will no longer be usable and all future behavior is undefined.
+   */
+  close(): Promise<void> {
+    if (!this._terminated) {
```
Comment (alexander-fenster) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk]: looks like the condition must be reversed

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 3]
```diff
@@ -254,21 +256,32 @@ export class {{ service.name }}Client {
...
+            this._descriptors.stream[methodName] ||
             this._descriptors.longrunning[methodName]
       );
+
+      this._innerApiCalls[methodName] = (
+        argument: {},
+        callOptions?: CallOptions,
+        callback?: APICallback
+      ) => {
+         if (!this._terminated) {
```
Comment (alexander-fenster) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk]: looks like the condition must be reversed

**Pull request merged**